Marlin/ultralcd.cpp

```diff
@@ -2847,101 +2853,94 @@ void lcd_update() {
       }
     #endif // ULTIPANEL
 
-    #if ENABLED(ENSURE_SMOOTH_MOVES) && ENABLED(ALWAYS_ALLOW_MENU)
-      #define STATUS_UPDATE_CONDITION planner.long_move()
-    #else
-      #define STATUS_UPDATE_CONDITION true
-    #endif
-    #if ENABLED(ENSURE_SMOOTH_MOVES) && DISABLED(ALWAYS_ALLOW_MENU)
-      #define LCD_HANDLER_CONDITION planner.long_move()
-    #else
-      #define LCD_HANDLER_CONDITION true
-    #endif
-
     // We arrive here every ~100ms when idling often enough.
     // Instead of tracking the changes simply redraw the Info Screen ~1 time a second.
     static int8_t lcd_status_update_delay = 1; // first update one loop delayed
-    if (STATUS_UPDATE_CONDITION &&
+    if (!lcd_status_update_delay--
       #if ENABLED(ULTIPANEL)
-        currentScreen == lcd_status_screen &&
+        && currentScreen == lcd_status_screen
       #endif
-        !lcd_status_update_delay--
     ) {
-      lcd_status_update_delay = 9;
+      lcd_status_update_delay = 3 + 9;
+      max_display_update_time--;
```

Comment:
`max_display_update_time` is decreased about 1/second this ensures, even if there was a long quadstep-move what interrupted randomly one partial screen update, the max_time will not be that high for ever.